components/virtual-node/sandbox-notary-impl/src/main/kotlin/net/corda/ledger/notary/plugin/factory/impl/PluggableNotaryClientFlowFactoryImpl.kt

```diff
@@ -53,8 +53,19 @@ class PluggableNotaryClientFlowFactoryImpl @Activate constructor(
             }
         }
 
+        if (pluginClass == null) {
+            throw CordaRuntimeException(
+                "Plugin class could not be retrieved. That either means there's no notary registered on the network, " +
+                        "or the provided notary party is not matching the one registered on the network."
```

Comment:
```suggestion
                "Plugin class not found for notary service $notaryService. This means that no notary service " +  
                        "matching this name has been registered on the network."
```

More concise, indicates the specific service not found, and avoids suggesting that there is only one service on the network.

(You might need to tweak the variable injection to get the x500 name from that object)